Framework/PythonInterface/plugins/algorithms/LoadNMoldyn4Ascii1D.py

```diff
@@ -0,0 +1,224 @@
+
+        for function in func_names:
+            prog_reporter.report('Loading {0} function'.format(function))
+            # Loads the two axes
+            y_axis = self.read_axis(function)
+            x_axis = self.read_axis(y_axis[3])
+
+            # Converts the x-axis units and sets all axis properties
+            x_axis = self.axis_conversion(x_axis[0], x_axis[1], y_axis[3])
+            y_data = y_axis[0]
+            y_name = y_axis[2]
+            x_data = x_axis[0]
+            x_unit = x_axis[1]
+            x_name = x_axis[2]
+
+            # Convolutes the data if required
+            if self.getPropertyValue('ResolutionConvolution') == 'TOSCA' and x_name == 'frequency':
+                resolutions = self.gaussians(x_data, self.TOSCA_resfunction)
+                y_data = self.convolutor(y_data, resolutions, x_data)
+                logger.information('Function ' + str(y_name) + ' will be convoluted')
+
+            # Create the workspace for function
+            ws_title = out_ws_name+'('+function+')'
+            CreateWorkspace(OutputWorkspace=ws_title, DataX=x_data, DataY=y_data,
+                            UnitX=x_unit, WorkspaceTitle=ws_title)
+
+            loaded_function_workspaces.append(ws_title)
+
+        if len(loaded_function_workspaces) == 0:
+            raise RuntimeError('Failed to load any functions for data')
+        GroupWorkspaces(InputWorkspaces=loaded_function_workspaces, OutputWorkspace=out_ws_name)
+        # Set the output workspace
+        self.setProperty('OutputWorkspace', out_ws_name)
+
+    def read_axis(self, func_name):
+        # Loads an axis file from the directory.
+
+        # Axis file loaded is directory\func_name.dat
+        # Returns a tuple of (Numpy array of data, unit, func_name, x-axis name)
+
+        file_name = os.path.join(self.data_directory, '{0}.dat'.format(func_name))
+        x_axis = 0
+        with open(file_name, 'rU') as f_handle:
+            while True:
+                line = f_handle.readline()
+                if not line:
+                    break
+                if len(line[0]) == 0:
+                    pass
+                # Parse metadata from file
+                elif line[0] == '#':
+                    func_match = FUNC_NAME_REGEX.match(line)
+                    if func_match:
+                        func_name = func_match.group(1)
+                    axis_match = AXIS_REGEX.match(line)
+                    if axis_match:
+                        x_axis = axis_match.group(1)
+                    unit_match = UNIT_REGEX.match(line)
+                    if unit_match:
+                        unit = unit_match.group(1)
+                    # Reads actual data from file
+                    slice_match = SLICE_HEADER_REGEX.match(line)
+                    if slice_match:
+                        length = int(slice_match.group(1))
+                        data = self.slice_read(f_handle, length)
```

Comment:
The code looks very good. Maybe here I'd suggest moving the long `elif` branch into a function of its own which produces the `func_name`, `unit`, etc. I just picked this up because all the functions are very short and/or trivial to follow (ideal) except this one which is a bit longer.